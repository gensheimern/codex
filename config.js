module.exports = {
	DB_HOST: process.env.DB_HOST || '159.65.115.221',
	DB_USER: process.env.DB_USER || 'root',
	DB_PASS: process.env.DB_PASS || 'teamcodex',
	DB_NAME: process.env.DB_NAME || 'lunch_planner',
	JWT_SECRET: process.env.JWT_SECRET || 'secret',
	MAIL_HOST: process.env.MAIL_HOST || '159.65.115.221',
	MAIL_USER: process.env.MAIL_USER || 'support@codex-team.de',
	MAIL_PASS: process.env.MAIL_PASS || 'teamcodex123',
};
