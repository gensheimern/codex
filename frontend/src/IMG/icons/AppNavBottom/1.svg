<?xml version="1.0" standalone="no"?>
<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 20010904//EN"
 "http://www.w3.org/TR/2001/REC-SVG-20010904/DTD/svg10.dtd">
<svg version="1.0" xmlns="http://www.w3.org/2000/svg"
 width="27.000000pt" height="27.000000pt" viewBox="0 0 27.000000 27.000000"
 preserveAspectRatio="xMidYMid meet">

<g transform="translate(0.000000,27.000000) scale(0.100000,-0.100000)"
fill="#000000" stroke="none">
<path d="M0 135 l0 -135 135 0 135 0 0 135 0 135 -135 0 -135 0 0 -135z m220
45 c0 -6 -35 -10 -85 -10 -50 0 -85 4 -85 10 0 6 35 10 85 10 50 0 85 -4 85
-10z m0 -50 c0 -6 -35 -10 -85 -10 -50 0 -85 4 -85 10 0 6 35 10 85 10 50 0
85 -4 85 -10z m-70 -50 c0 -5 -22 -10 -50 -10 -27 0 -50 5 -50 10 0 6 23 10
50 10 28 0 50 -4 50 -10z"/>
</g>
</svg>
