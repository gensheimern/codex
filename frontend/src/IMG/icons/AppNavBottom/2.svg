<?xml version="1.0" standalone="no"?>
<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 20010904//EN"
 "http://www.w3.org/TR/2001/REC-SVG-20010904/DTD/svg10.dtd">
<svg version="1.0" xmlns="http://www.w3.org/2000/svg"
 width="27.000000pt" height="27.000000pt" viewBox="0 0 27.000000 27.000000"
 preserveAspectRatio="xMidYMid meet">

<g transform="translate(0.000000,27.000000) scale(0.100000,-0.100000)"
fill="#000000" stroke="none">
<path d="M0 135 l0 -135 135 0 135 0 0 135 0 135 -24 0 c-18 0 -25 -6 -28 -22
-5 -34 -38 -37 -38 -4 0 33 -16 34 -24 1 -4 -14 -13 -25 -21 -25 -8 0 -17 11
-21 25 -8 33 -24 32 -24 -1 0 -33 -33 -30 -38 4 -3 16 -10 22 -28 22 l-24 0 0
-135z m75 55 c3 -5 -3 -10 -14 -10 -12 0 -21 5 -21 10 0 6 6 10 14 10 8 0 18
-4 21 -10z m85 0 c0 -5 -9 -10 -21 -10 -11 0 -17 5 -14 10 3 6 13 10 21 10 8
0 14 -4 14 -10z m75 0 c4 -6 -5 -10 -20 -10 -15 0 -24 4 -20 10 3 6 12 10 20
10 8 0 17 -4 20 -10z m-155 -50 c0 -5 -9 -10 -20 -10 -11 0 -20 5 -20 10 0 6
9 10 20 10 11 0 20 -4 20 -10z m80 0 c0 -5 -9 -10 -20 -10 -11 0 -20 5 -20 10
0 6 9 10 20 10 11 0 20 -4 20 -10z m75 0 c4 -6 -5 -10 -20 -10 -15 0 -24 4
-20 10 3 6 12 10 20 10 8 0 17 -4 20 -10z m-155 -50 c0 -5 -9 -10 -20 -10 -11
0 -20 5 -20 10 0 6 9 10 20 10 11 0 20 -4 20 -10z m80 0 c0 -5 -9 -10 -20 -10
-11 0 -20 5 -20 10 0 6 9 10 20 10 11 0 20 -4 20 -10z m75 0 c4 -6 -5 -10 -19
-10 -14 0 -26 5 -26 10 0 6 9 10 19 10 11 0 23 -4 26 -10z"/>
<path d="M60 250 c0 -11 5 -20 10 -20 6 0 10 9 10 20 0 11 -4 20 -10 20 -5 0
-10 -9 -10 -20z"/>
<path d="M132 250 c0 -14 2 -19 5 -12 2 6 2 18 0 25 -3 6 -5 1 -5 -13z"/>
<path d="M190 250 c0 -11 5 -20 10 -20 6 0 10 9 10 20 0 11 -4 20 -10 20 -5 0
-10 -9 -10 -20z"/>
</g>
</svg>
