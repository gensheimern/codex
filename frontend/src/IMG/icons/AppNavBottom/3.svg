<?xml version="1.0" standalone="no"?>
<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 20010904//EN"
 "http://www.w3.org/TR/2001/REC-SVG-20010904/DTD/svg10.dtd">
<svg version="1.0" xmlns="http://www.w3.org/2000/svg"
 width="38.000000pt" height="38.000000pt" viewBox="0 0 38.000000 38.000000"
 preserveAspectRatio="xMidYMid meet">

<g transform="translate(0.000000,38.000000) scale(0.100000,-0.100000)"
fill="#000000" stroke="none">
<path d="M0 313 c1 -65 1 -67 14 -38 16 36 61 80 96 94 20 8 13 10 -42 10
l-68 1 0 -67z"/>
<path d="M270 369 c35 -14 80 -58 96 -94 13 -29 13 -27 14 38 l0 67 -67 -1
c-56 0 -63 -2 -43 -10z"/>
<path d="M1 68 l-1 -68 68 0 c64 1 66 1 37 14 -36 16 -80 61 -94 96 -8 20 -10
13 -10 -42z"/>
<path d="M370 115 c0 -22 -60 -86 -95 -101 -29 -13 -27 -13 38 -14 l67 0 0 65
c0 36 -2 65 -5 65 -3 0 -5 -7 -5 -15z"/>
</g>
</svg>
