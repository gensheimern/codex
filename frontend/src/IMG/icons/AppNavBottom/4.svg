<?xml version="1.0" standalone="no"?>
<!DOCTYPE svg PUBLIC "-//W3C//DTD SVG 20010904//EN"
 "http://www.w3.org/TR/2001/REC-SVG-20010904/DTD/svg10.dtd">
<svg version="1.0" xmlns="http://www.w3.org/2000/svg"
 width="27.000000pt" height="27.000000pt" viewBox="0 0 27.000000 27.000000"
 preserveAspectRatio="xMidYMid meet">

<g transform="translate(0.000000,27.000000) scale(0.100000,-0.100000)"
fill="#000000" stroke="none">
<path d="M1 135 l1 -135 134 0 134 0 0 135 0 135 -135 0 -135 0 1 -135z"/>
</g>
</svg>
