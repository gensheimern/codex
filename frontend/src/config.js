const config = {
	debug: process.env.REACT_APP_DEBUG === 'TRUE',
	apiPath: process.env.REACT_APP_API_PATH,
	wsPath: process.env.REACT_APP_WS_PATH,
	basePath: process.env.REACT_APP_BASE_PATH,
}

export default config;