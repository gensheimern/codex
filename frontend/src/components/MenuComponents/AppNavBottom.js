import React, {Component} from 'react';
import {BottomNavigation, BottomNavigationItem} from 'material-ui/BottomNavigation';
import Paper from 'material-ui/Paper';
import IconAdd from 'material-ui/svg-icons/content/add-circle';
import IconProfile from 'material-ui/svg-icons/social/person';
import IconCalendar from 'material-ui/svg-icons/action/event';
import IconFeed from 'material-ui/svg-icons/action/assignment';
import Notification from 'material-ui/svg-icons/social/notifications';
import { withRouter } from 'react-router-dom';
import notificationChecker from '../notification/notificationChecker';
import Badge from 'material-ui/Badge';
import config from '../../config';
import './sidebars.css';

const FeedIcon = <IconFeed/>;
const CalendarIcon = <IconCalendar/>;
const AddIcon = <IconAdd/>;
const ProfileIcon = <IconProfile />;

/**
 * A simple example of `BottomNavigation`, with three labels and icons
 * provided. The selected `BottomNavigationItem` is determined by application
 * state (for instance, by the URL).
 */
class AppNavBottom extends Component {
	state = {
		selectedIndex: 0,
		newNotifications: 0,
		unseenNotifications: 0,
		unsubscribed: false,
	};

	select = (index) => {
		this.readNotifications();

		this.setState({selectedIndex: index})
		this.props.changeContent(index);
	};

	componentDidMount() {
		notificationChecker.enable('notification', () => {
			if (!this.state.unsubscribed) {
				this.setState((prevState, props) => ({
					newNotifications: prevState.newNotifications + 1,
				}));
			}
		});

		this.loadUnseenNotifications();
	}

	loadUnseenNotifications() {
		fetch(config.apiPath + "/user/me/notifications/unseen", {
						method: 'GET',
						headers: {
				'Content-Type': 'application/json',
				'X-Access-Token': localStorage.getItem('apiToken'),
						}
		})
		.then((res) => {
						if(!res.ok) {
								throw new Error("Response not ok.");
						} else if(res.status !== 200) {
								throw new Error("An error occured.");
						}
						return res;
		})
		.then(res => res.json())
		.then((res) => {
						this.setState({
				unseenNotifications: res.unseenNotifications,
			});
				}).catch((error) => {
			this.setState({
				unseenNotifications: 0,
			});
				});
	}

	componentWillUnmount() {
		this.setState({
			unsubscribed: true,
		});
		notificationChecker.disable('notification');
	}

	readNotifications = () => {
		this.setState({
			newNotifications: 0,
			unseenNotifications: 0,
		});
	}

	render() {
		const notificationIcon = this.state.newNotifications + this.state.unseenNotifications > 0
			? 	(<Badge
					badgeContent={this.state.newNotifications + this.state.unseenNotifications}
					badgeStyle={{
						backgroundColor: 'red',
						color: 'white',
						top: 1,
						right: 12,
						fontSize: '10px',
						width: '16px',
						height: '16px',
						padding : '0px !important',
						margin : '0px !important',
					}}
					style={{
						padding : "0px !important",
					}}
				>
					<Notification onClick={() => {
						this.readNotifications();
						this.props.history.push('/notifications');
					}}
					className="BadgeNotif"
					style={{
						cursor: 'pointer',
						padding : '0px !important',
						top: 1,
						right: 12,
					}}
					/>
				</Badge>)
			:	(<Notification onClick={() => {
					this.readNotifications();
					this.props.history.push('/notifications');
				}}/>);

		const itemStyle = {
			minWidth: '56px',
		};

		return (
			<Paper className="navbarbottom" zDepth={1}>
				<BottomNavigation selectedIndex={this.props.index}>
					<BottomNavigationItem
						icon={FeedIcon}
						onClick={() => this.props.history.push('/feed')}
						style={itemStyle}
					/>
					<BottomNavigationItem
						icon={CalendarIcon}
						onClick={() => this.props.history.push('/personal')}
						style={itemStyle}
					/>
					<BottomNavigationItem
						icon={AddIcon}
						onClick={() => this.props.history.push('/addevent')}
						style={{
							...itemStyle,
							padding: '0px !important',
						}}
					/>
					<BottomNavigationItem
						icon={notificationIcon}
						onClick={() => this.props.history.push('/notifications')}
						style={itemStyle}
					/>
					<BottomNavigationItem
						icon={ProfileIcon}
						onClick={() => this.props.history.push('/profile')}
						style={itemStyle}
					/>
				</BottomNavigation>
			</Paper>
		);
	}
}

export default withRouter(AppNavBottom);
