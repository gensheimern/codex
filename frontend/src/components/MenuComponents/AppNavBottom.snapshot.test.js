/**
 * @author:Stella Neser
 *  Komponente ist ein Snapshottest
 */
import React from 'react';
import renderer from'react-test-renderer';
import AppNavBottom from './AppNavBottom';
import config from '../../config';
import {MemoryRouter} from 'react-router-dom';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

const fetchMock = require('fetch-mock');

beforeEach(() => {
    window.localStorage = {getItem: () => ""};
    fetchMock.get(config.apiPath +'/user/me/notifications/unseen', {
        unseenNotifications: 4,
    
    });
});

describe('AppNavBottom Snapshot', () => {
    test('renders', () =>{
        const component = renderer.create(
            <MemoryRouter>
                 <MuiThemeProvider>
                    <AppNavBottom/>
                </MuiThemeProvider>
            </MemoryRouter>
        );
        let tree= component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});