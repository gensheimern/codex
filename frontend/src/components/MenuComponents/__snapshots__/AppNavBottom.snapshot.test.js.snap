// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppNavBottom Snapshot renders 1`] = `
<div
  className="navbarbottom"
  style={
    Object {
      "WebkitTapHighlightColor": "rgba(0,0,0,0)",
      "backgroundColor": "#ffffff",
      "borderRadius": 2,
      "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
             0 1px 4px rgba(0, 0, 0, 0.12)",
      "boxSizing": "border-box",
      "color": "rgba(0, 0, 0, 0.87)",
      "fontFamily": "Roboto, sans-serif",
      "muiPrepared": true,
      "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
    }
  }
>
  <div
    style={
      Object {
        "backgroundColor": "#ffffff",
        "display": "flex",
        "height": 56,
        "justifyContent": "center",
        "muiPrepared": true,
        "position": "relative",
        "width": "100%",
      }
    }
  >
    <button
      disabled={undefined}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      selected={false}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "background": "none",
          "border": 10,
          "boxSizing": "border-box",
          "cursor": "pointer",
          "display": "inline-block",
          "flex": "1",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": "inherit",
          "fontWeight": "inherit",
          "margin": 0,
          "maxWidth": 168,
          "minWidth": "56px",
          "muiPrepared": true,
          "outline": "none",
          "padding": 0,
          "paddingBottom": 10,
          "paddingLeft": 12,
          "paddingRight": 12,
          "paddingTop": 8,
          "position": "relative",
          "textDecoration": "none",
          "transition": "padding-top 0.3s",
          "verticalAlign": null,
        }
      }
      tabIndex={0}
      type="button"
    >
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <svg
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "display": "block",
              "fill": "rgba(0, 0, 0, 0.54)",
              "height": 24,
              "muiPrepared": true,
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
              "width": "100%",
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M19 3h-4.18C14.4 1.84 13.3 1 12 1c-1.3 0-2.4.84-2.82 2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zm-7 0c.55 0 1 .45 1 1s-.45 1-1 1-1-.45-1-1 .45-1 1-1zm2 14H7v-2h7v2zm3-4H7v-2h10v2zm0-4H7V7h10v2z"
          />
        </svg>
        <div
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.54)",
              "fontSize": 12,
              "muiPrepared": true,
              "transition": "color 0.3s, font-size 0.3s",
            }
          }
        />
      </div>
    </button>
    <button
      disabled={undefined}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      selected={false}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "background": "none",
          "border": 10,
          "boxSizing": "border-box",
          "cursor": "pointer",
          "display": "inline-block",
          "flex": "1",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": "inherit",
          "fontWeight": "inherit",
          "margin": 0,
          "maxWidth": 168,
          "minWidth": "56px",
          "muiPrepared": true,
          "outline": "none",
          "padding": 0,
          "paddingBottom": 10,
          "paddingLeft": 12,
          "paddingRight": 12,
          "paddingTop": 8,
          "position": "relative",
          "textDecoration": "none",
          "transition": "padding-top 0.3s",
          "verticalAlign": null,
        }
      }
      tabIndex={0}
      type="button"
    >
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <svg
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "display": "block",
              "fill": "rgba(0, 0, 0, 0.54)",
              "height": 24,
              "muiPrepared": true,
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
              "width": "100%",
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
          />
        </svg>
        <div
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.54)",
              "fontSize": 12,
              "muiPrepared": true,
              "transition": "color 0.3s, font-size 0.3s",
            }
          }
        />
      </div>
    </button>
    <button
      disabled={undefined}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      selected={false}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "background": "none",
          "border": 10,
          "boxSizing": "border-box",
          "cursor": "pointer",
          "display": "inline-block",
          "flex": "1",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": "inherit",
          "fontWeight": "inherit",
          "margin": 0,
          "maxWidth": 168,
          "minWidth": "56px",
          "muiPrepared": true,
          "outline": "none",
          "padding": "0px !important",
          "paddingBottom": 10,
          "paddingLeft": 12,
          "paddingRight": 12,
          "paddingTop": 8,
          "position": "relative",
          "textDecoration": "none",
          "transition": "padding-top 0.3s",
          "verticalAlign": null,
        }
      }
      tabIndex={0}
      type="button"
    >
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <svg
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "display": "block",
              "fill": "rgba(0, 0, 0, 0.54)",
              "height": 24,
              "muiPrepared": true,
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
              "width": "100%",
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm5 11h-4v4h-2v-4H7v-2h4V7h2v4h4v2z"
          />
        </svg>
        <div
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.54)",
              "fontSize": 12,
              "muiPrepared": true,
              "transition": "color 0.3s, font-size 0.3s",
            }
          }
        />
      </div>
    </button>
    <button
      disabled={undefined}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      selected={false}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "background": "none",
          "border": 10,
          "boxSizing": "border-box",
          "cursor": "pointer",
          "display": "inline-block",
          "flex": "1",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": "inherit",
          "fontWeight": "inherit",
          "margin": 0,
          "maxWidth": 168,
          "minWidth": "56px",
          "muiPrepared": true,
          "outline": "none",
          "padding": 0,
          "paddingBottom": 10,
          "paddingLeft": 12,
          "paddingRight": 12,
          "paddingTop": 8,
          "position": "relative",
          "textDecoration": "none",
          "transition": "padding-top 0.3s",
          "verticalAlign": null,
        }
      }
      tabIndex={0}
      type="button"
    >
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <svg
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "display": "block",
              "fill": "rgba(0, 0, 0, 0.54)",
              "height": 24,
              "muiPrepared": true,
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
              "width": "100%",
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M12 22c1.1 0 2-.9 2-2h-4c0 1.1.89 2 2 2zm6-6v-5c0-3.07-1.64-5.64-4.5-6.32V4c0-.83-.67-1.5-1.5-1.5s-1.5.67-1.5 1.5v.68C7.63 5.36 6 7.92 6 11v5l-2 2v1h16v-1l-2-2z"
          />
        </svg>
        <div
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.54)",
              "fontSize": 12,
              "muiPrepared": true,
              "transition": "color 0.3s, font-size 0.3s",
            }
          }
        />
      </div>
    </button>
    <button
      disabled={undefined}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      selected={false}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "background": "none",
          "border": 10,
          "boxSizing": "border-box",
          "cursor": "pointer",
          "display": "inline-block",
          "flex": "1",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": "inherit",
          "fontWeight": "inherit",
          "margin": 0,
          "maxWidth": 168,
          "minWidth": "56px",
          "muiPrepared": true,
          "outline": "none",
          "padding": 0,
          "paddingBottom": 10,
          "paddingLeft": 12,
          "paddingRight": 12,
          "paddingTop": 8,
          "position": "relative",
          "textDecoration": "none",
          "transition": "padding-top 0.3s",
          "verticalAlign": null,
        }
      }
      tabIndex={0}
      type="button"
    >
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <svg
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.87)",
              "display": "block",
              "fill": "rgba(0, 0, 0, 0.54)",
              "height": 24,
              "muiPrepared": true,
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
              "width": "100%",
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
          />
        </svg>
        <div
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.54)",
              "fontSize": 12,
              "muiPrepared": true,
              "transition": "color 0.3s, font-size 0.3s",
            }
          }
        />
      </div>
    </button>
  </div>
</div>
`;
